tests/state/test_get_list_state.py

```diff
@@ -0,0 +1,389 @@
+
+import pytest
+from click.testing import CliRunner
+
+from cli.constants import (
+    FAILED_EXIT_FROM_COMMAND_CODE,
+    NODE_IP_ADDRESS_FOR_TESTING,
+    PASSED_EXIT_FROM_COMMAND_CODE,
+)
+from cli.entrypoint import cli
+from cli.utils import dict_to_pretty_json
+
+ADDRESS_WITH_STATE = '0000000000000000000000000000000000000000000000000000000000000000000001'
+
+
+def test_get_list_state():
+    """
+    Case: get a list of states.
+    Expect: states are returned.
+    """
+    runner = CliRunner()
+    result = runner.invoke(cli, [
+        'state',
+        'get-list',
+        '--node-url',
+        NODE_IP_ADDRESS_FOR_TESTING,
+    ])
+
+    result_data = json.loads(result.output).get('result')[0]
+    assert PASSED_EXIT_FROM_COMMAND_CODE == result.exit_code
+    assert result_data
+
+
+def test_get_list_state_with_all_parameters():
+    """
+    Case: get a list of states by account address, starting address, limit, head, reverse.
+    Expect: state is returned.
+    """
+    head = '56100bf24eed12d2f72fe3c3ccf75fe2f53d87c224d9dda6fb98a1411070b06a' \
+           '40fcf97fccc61cb9c88442953af6ae50344ad7773f1becc6bae108443c18c551'
+
+    runner = CliRunner()
+    result = runner.invoke(cli, [
+        'state',
+        'get-list',
+        '--address',
+        ADDRESS_WITH_STATE,
+        '--start',
+        ADDRESS_WITH_STATE,
+        '--limit',
+        1,
+        '--head',
+        head,
+        '--node-url',
+        NODE_IP_ADDRESS_FOR_TESTING,
+        '--reverse',
+    ])
+
+    result_data = json.loads(result.output).get('result')
+
+    assert PASSED_EXIT_FROM_COMMAND_CODE == result.exit_code
+    assert result_data
+
+
+def test_get_list_state_with_address():
+    """
+    Case: get a state by its address.
+    Expect: state is returned.
+    """
+    runner = CliRunner()
+    result = runner.invoke(cli, [
+        'state',
+        'get-list',
+        '--address',
+        ADDRESS_WITH_STATE,
+        '--node-url',
+        NODE_IP_ADDRESS_FOR_TESTING,
+    ])
+
+    result_data = json.loads(result.output).get('result')[0]
+
+    assert PASSED_EXIT_FROM_COMMAND_CODE == result.exit_code
+    assert result_data
+
+
+def test_get_list_state_with_invalid_address():
+    """
+    Case: get a state by its invalid address.
+    Expect: the following address is invalid error message.
+    """
+    invalid_address = '044c7db163cf2'
+
+    runner = CliRunner()
+    result = runner.invoke(cli, [
+        'state',
+        'get-list',
+        '--address',
+        invalid_address,
+        '--node-url',
+        NODE_IP_ADDRESS_FOR_TESTING,
+    ])
+
+    expected_error_message = {
+        'errors': {
+            'address': [
+                f'The following address `{invalid_address}` is invalid.',
+            ],
+        },
+    }
+
+    assert FAILED_EXIT_FROM_COMMAND_CODE == result.exit_code
+    assert dict_to_pretty_json(expected_error_message) in result.output
+
+
+def test_get_list_state_with_non_existing_address():
```

Comment:
Make one test case using `@pytest.mark.parametrize` for tests with an invalid argument (address, start), they have the same error message `'Block not found.'`.

Reply:
@anastasiia-bilova, done.